utils/task_tester/inginious-submission-anonymizer

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# This file is part of INGInious. See the LICENSE and the COPYRIGHTS files for
+# more information about the licensing of this file.
+
+from hashlib import sha256
+import tempfile
+import tarfile
+import shutil
+import sys
+import os
+import re
+
+#from inginious.common.filesystems.local import LocalFSProvider
+
+def anonymize(courseid:str, path: str):
+    cwd = os.getcwd()
+
+    """
+     10     -e "s/'@email': (.*)$/'@email': test@test/g"\                                                     
+     11     -e "s/'@username': (.*)$/'@username': test/g"\                                                    
+    """
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        with tarfile.open(path, 'r') as tar:
+            while x := tar.next():
+                if x is None: break
+
+                if not x.isfile(): continue
+                filename, extension = os.path.splitext(x.name)
+                if extension != ".test": continue
+
+                task, _, filename = filename.partition('/')
+                new_name = '_'.join(filename.split('/'))
+                a = '{root}/{courseid}/{task}/test/{h}.test'.format(
+                        root=cwd,
+                        courseid=courseid,
+                        task=task,
+                        h=sha256(new_name.encode('utf-8')).hexdigest()
+                )
+                tar.extract(x, path=tmpdir)
+
+                with open('%s/%s' % (tmpdir, x.name), 'r') as fd:
+                    content = fd.read()
+
+                # remove identifiable fields
+                new_content = re.sub(r'(user_ip|_id|archive|grade|submitted_on|time):(.*)', '', content)
```

Comment:
I guess that'll be more modular to put all these strings as a list of 2-tuples, as you may add more metadata fields in the future, and replace those lines by a for loop.

Reply:
See https://github.com/UCL-INGI/INGInious/pull/861/commits/35fbb2db46b909b6bf9cd5cdb97e19766dc61ab3